sql/src/main/java/io/crate/protocols/postgres/BatchPortal.java

```diff
@@ -126,7 +126,11 @@ public void execute(ResultReceiver resultReceiver, int maxRows) {
     public void sync(Planner planner, StatsTables statsTables, CompletionListener listener) {
         BatchCompletionListener batchCompletionListener = new BatchCompletionListener(analysis.size(), listener);
         for (int i = 0; i < analysis.size(); i++) {
-            UUID jobId = UUID.randomUUID();
+            UUID jobId = sessionData.getJobId();
+            if (i > 0) {
+               jobId = UUID.randomUUID();
```

Comment:
Because in SQLOperations parse method whenever we parse a new statement we create a jobId and put in in statsTables. When we get a second parse in the same context the we understand that it's actually a batch operation but we already have an entry in statsTables for the first statement.